lib/clients/BaseClient.py

```diff
@@ -70,6 +71,28 @@ def __init__(self, operation_name, configuration, directory_persistent, director
         self.__mounted_devices = []
         self.__devices = {}
 
+    def _get_credentials_from_credhub(self, configuration):
+        access_token = self.__getAccessToken(configuration)
+        params = {'name': configuration['credhub_key']}
+        headers = {'content-type': 'application/json',
+                   'authorization': 'bearer '+ access_token }
+        response = requests.get(url = configuration['credhub_url'].rstrip('/') + '/v1/data', headers=headers, params= params, verify=False)
+        credentials = response.json()
+        return credentials['data'][0]['value']
+
+    def __getAccessToken(self, configuration):
+        payload = {
+            'grant_type' : 'password',
```

Comment:
Any specific reason for selecting password grant type?

Reply:
part of the OAUTH spec - https://docs.cloudfoundry.org/api/uaa/version/4.14.0/index.html#password-grant

Reply:
oAuth spec has many other grants why was password grant picked up for implementation?

Reply:
This is the std grant type for service accounts to exchange creds for access tokens.